### PR TITLE
Issue 118 self updater release tag check fix

### PR DESCRIPTION
**Pull request by rpsu**
- Added info to README how to update the local-docker itself
- Update self-update -command info (even if not used)
- Whitespace + indenting, minor UI text updates
- use **releases**, not tags
- Use latest release or a release with a given name

### CODE REVIEW

**Comment by rpsu** [comment]
Fixes #118

File: docker/scripts/self-update.sh [hunk 1]
```diff
@@ -4,63 +4,72 @@
...
-  fi
+# When no tag is provided we'll fallback to use the 'latest'.
+TAG=${1:-latest}
+TAG_PROVIDED=1
+if [ -z "$1" ];then
+    TAG_PROVIDED=
 fi
 
+# Check the tag exists if one is provided.
+[ ! -z "$TAG_PROVIDED" ] && echo "Verifying release ${TAG} exists, please wait..."
```
Comment (kekkis) [inline docker/scripts/self-update.sh]: Also `[ -n "$TAG_PROVIDED" ]` and `[ "$TAG_PROVIDED" ]` works.

File: docker/scripts/self-update.sh [hunk 2]
```diff
@@ -4,63 +4,72 @@
...
 
 # Curl creates an ASCII file out of 404 response. Let's see what we have in the file.
-INFO=$(file -b $DIR/${TAG}.tar.gz | cut -d' ' -f1)
-
+INFO=$(file -b $DIR/$TEMP_FILENAME | cut -d' ' -f1)
 if [ "$INFO" != "gzip" ]; then
-  echo -e "${Red}ERROR: Specifidd tag not found.${Color_Off}"
-  rm -rf $DIR
-  return 1
+    echo -e "${Red}ERROR: Download the the requested release failed.${Color_Off}"
```
Comment (kekkis) [inline docker/scripts/self-update.sh]: Typofix.
```suggestion
    echo -e "${Red}ERROR: Downloading the requested release failed.${Color_Off}"
```

File: docker/scripts/self-update.sh [hunk 3]
```diff
@@ -4,63 +4,72 @@
...
+    rm -rf $DIR
+    return 1
 fi
 
-cd $DIR
-tar xvzf ${TAG}.tar.gz
+tar xzf $DIR/$TEMP_FILENAME -C $DIR
```
Comment (kekkis) [inline docker/scripts/self-update.sh]: Should the variables be fully quoted? E.g.
```suggestion
tar xzf "$DIR/$TEMP_FILENAME" -C "$DIR"
```

File: docker/scripts/self-update.sh [hunk 4]
```diff
@@ -4,63 +4,72 @@
...
 SUBDIR=$(ls |grep local-docker)
 LIST=" .editorconfig .env.example .env.local.example .gitignore.example ./.github ./docker ./git-hooks ld.sh"
 for FILE in $LIST; do
-  cp -fvR $SUBDIR/$FILE ../ &>/dev/null 2>&1
+    cp -fr $DIR/$SUBDIR/$FILE . 2>/dev/null
```
Comment (kekkis) [inline docker/scripts/self-update.sh]: Here as well, maybe use quotes.
```suggestion
    cp -fr "$DIR/$SUBDIR/$FILE" . 2>/dev/null
```

File: docker/scripts/self-update.sh [hunk 5]
```diff
@@ -4,63 +4,72 @@
...
+if [ -z "$TAG_PROVIDED" ]; then
+     # Latest git tags is the first one in the file.
+    echo -e "Release name : ${BGreen} $RELEASE_NAME${Color_Off}"
+    echo -e "Published    : ${BGreen} $RELEASE_PUBLISHED${Color_Off}"
+    echo -e "Release page : ${BGreen} $RELEASE_PAGE${Color_Off}"
+    echo -e "Release info : "
+    echo -e "${BGreen}$RELEASE_BODY${Color_Off}"
+    echo "Downloading release from $RELEASE_TARBALL, please wait..."
+    # -L to follow redirects
+    curl -L -s -o $DIR/$TEMP_FILENAME $RELEASE_TARBALL
```
Comment (kekkis) [inline docker/scripts/self-update.sh]:
```suggestion
    curl -L -s -o "$DIR/$TEMP_FILENAME" "$RELEASE_TARBALL"
```

File: docker/scripts/self-update.sh [hunk 6]
```diff
@@ -4,63 +4,72 @@
...
-  # -L to follow redirects
-  echo "Acquiring version from : $URL"
-  curl -L -s -o $DIR/${TAG}.tar.gz $URL
-
-else
-  URL="https://api.github.com/repos/Exove/local-docker/tarball/${TAG}"
-  # -L to follow redirects
-  curl -L -s -o $DIR/${TAG}.tar.gz $URL
-  ls -lah $DIR
+TEMP_FILENAME=release-${RELEASE_NAME}.tar.gz
```
Comment (kekkis) [inline docker/scripts/self-update.sh]: if `RELEASE_NAME` can have spaces, quoting the variable in some way might make sense?

File: docker/scripts/self-update.sh [hunk 7]
```diff
@@ -4,63 +4,72 @@
...
+tar xzf $DIR/$TEMP_FILENAME -C $DIR
 SUBDIR=$(ls |grep local-docker)
 LIST=" .editorconfig .env.example .env.local.example .gitignore.example ./.github ./docker ./git-hooks ld.sh"
 for FILE in $LIST; do
-  cp -fvR $SUBDIR/$FILE ../ &>/dev/null 2>&1
+    cp -fr $DIR/$SUBDIR/$FILE . 2>/dev/null
 done
-cd ..
+
 rm -rf $DIR
```
Comment (kekkis) [inline docker/scripts/self-update.sh]:
```suggestion
rm -rf "$DIR"
```

Also, let's hope `$DIR` never evaluates to `/`

**Pull request merged**